crates/ruff_python_formatter/src/other/except_handler_except_handler.rs

```diff
@@ -11,6 +15,34 @@ impl FormatNodeRule<ExceptHandlerExceptHandler> for FormatExceptHandlerExceptHan
         item: &ExceptHandlerExceptHandler,
         f: &mut PyFormatter,
     ) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let ExceptHandlerExceptHandler {
+            range: _,
+            type_,
+            name,
+            body,
+        } = item;
+
+        let comments_info = f.context().comments().clone();
+        let dangling_comments = comments_info.dangling_comments(item.as_any_node_ref());
+
+        write!(f, [text("except")])?;
+
+        if let Some(type_) = type_ {
+            write!(
+                f,
+                [space(), type_.format().with_options(Parenthesize::IfBreaks)]
+            )?;
+            if let Some(name) = name {
+                write!(f, [space(), text("as"), space(), name.format()])?;
+            }
+        }
+        write!(
+            f,
+            [
+                text(":"),
+                trailing_comments(dangling_comments),
+                block_indent(&body.format())
+            ]
+        )
     }
 }
```

Comment:
You can override `fmt_dangling_comments` with a stub implementation that always returns `Ok(())`, similar to what we do in `FormatTryStmt`

Reply:
You do 

```rust
    fn fmt_dangling_comments(
        &self,
        _node: &ExprAttribute,
        _f: &mut PyFormatter,
    ) -> FormatResult<()> {
        // handle in `fmt_fields`
        Ok(())
    }
```

in the `FormatNodeRule` implementation of `FormatExceptHandlerExceptHandler`.

Reply:
yes i understand _what you were asking for but not _why it's needed

Reply:
Technically it is not needed because `SourceComment`s track whether they have been formatted. It is mainly a perf improvement to avoid the additional `Comments` hashmap lookup when we already know that we've formatted all comments.